src/server/game/Entities/Object/Object.cpp

```diff
@@ -1147,7 +1147,12 @@ bool WorldObject::IsWithinLOSInMap(const WorldObject* obj) const
         return false;
 
     float ox, oy, oz;
-    obj->GetPosition(ox, oy, oz);
+    float angle = obj->GetAngle(GetPositionX(), GetPositionY());
+    float range = std::min(GetObjectSize(), GetExactDist(obj->GetPositionX(), obj->GetPositionY(), obj->GetPositionZ()));
```

Comment:
.debug LoS says I am in LoS with it, wondering why https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Spell.cpp#L4843
it checks if target is in los with caster